src/Umbraco.Core/Migrations/MigrationPlan.cs

```diff
@@ -287,13 +295,15 @@ public string Execute(IScope scope, string fromState, IMigrationBuilder migratio
             logger.Info<MigrationPlan>("At {OrigState}", string.IsNullOrWhiteSpace(origState) ? "origin": origState);
 
             if (!_transitions.TryGetValue(origState, out var transition))
-                throw new Exception($"Unknown state \"{origState}\".");
+                ThrowOnUnknownInitialState(origState);
```

Comment:
Why is this not just an ArgumentException? - I don't like we throw `Exception` directly (I know this is not new in this PR)

Reply:
`ThrowOnUnknownInitialState` is overriden by `UmbracoPlan` to give a more detailed error message, which is then less about "invalid argument" as about "invalid or not supported operation". Proposing to throw `InvalidOperationException`. Thoughts?

Reply:
`InvalidOperationException` is a good fit, if this is more than invalid arguments :)